spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/FastForwardBranchProcedure.java

```diff
@@ -77,9 +77,9 @@ public InternalRow[] call(InternalRow args) {
     return modifyIcebergTable(
         tableIdent,
         table -> {
-          long currentRef = table.currentSnapshot().snapshotId();
+          long currentRef = table.snapshot(source).snapshotId();
           table.manageSnapshots().fastForwardBranch(source, target).commit();
```

Comment:
Another thing to fix....I think when I originally implemented the fastForward operation I switched the definitions of source/target in my head so the API is confusing (in the API, target means the branch that is actually being fast forwarded, source is where target will be moved). That should probably be reversed so target is actually the target to which source will be moved.

I *think* we should be able to safely just rename the parameters in the API and update the javadoc. Luckily both are String parameters.

Reply:
I too got confused today when I checked replace branch API vs this procedure variables. Source and target is reversed. 

The procedure's named arguments of 'branch' and 'to' is proper. It is like fast forward branch x to y. 
Only thing is these internal variables in this procedure is reversed.  I think we can rename it in this PR. 

`ManageSnapshots.replaceBranch` and `ManageSnapshots.fastForwardBranch` seems to have a correct naming IMO

Reply:
I also agree it is confusing. It would be nice to rename in a follow-up PR or here. It seems like `source` represents a branch name in this case.

Reply:
I put up a separate PR here https://github.com/apache/iceberg/pull/9134